developer/src/kmc-ldml/src/compiler/compiler.ts

```diff
@@ -291,10 +291,14 @@ export class LdmlKeyboardCompiler implements KeymanCompiler {
   }
 
   /**
-   * Runs any linter steps
+   * Runs any linter steps, adding hints to the callbacks as needed
    * @internal
+   * @returns true unless there was a linter failure.
    */
   private async lint(source: LDMLKeyboardXMLSourceFile, kmx: KMXPlus.KMXPlusFile) : Promise<boolean> {
+    if (!kmx || !source) {
+      return false;
```

Comment:
So... if there's no source or no kmx, that's a linter failure?

Reply:
Well, I see that it isn't a linter failure just a broken pre-condition so bail returning false.